custom_components/eo_mini/api.py

```diff
@@ -97,6 +97,45 @@ async def async_post_enable(self, address) -> list[dict]:
             headers={"Content-Type": "application/x-www-form-urlencoded"},
         )
 
+    async def async_charge_mode_disable(
+        self, mode: str, charge_options: dict
```

Comment:
Not sure passing this dict in is very usable - especially since all the callsites are just passing in the original charge options from the controller anyway. I'd recommend:
1. not passing the dictionary in at all.
2. Copying the dictionary from `self.charge_options` before mutating it, so we know the value we query is definitely the value from the EO servers, and not the value we sent last time.

Reply:
(this is for all these methods)

Reply:
I passed in the dictionary to avoid a circular dependency - `self.charge_options` is in the class `EODataUpdateCoordinator` from `__init__.py` and `__init__.py` imports `EOApiClient` from `api.py` so I didn't see how I could access it within `api.py`
If you could show me example code of how to do this, I can update it, but I couldn't work it out myself

Reply:
Yeah that makes sense. I guess we should be passing in the dictionary we want to send in that case.